### ISSUE TITLE
Add volume test cases for Windows

### ISSUE THREAD

**Issue by PatrickLang** [kubernetes/kubernetes#69904]
We need test cases for Windows volume mapping permissions (read-only / read-write) instead of Linux permissions. They should only run with the ginkgo focus `--node-os-distro=windows` set.

The test cases needed are:
1. Create a pod with a host mapped volume mapped read-only. Files should be able to be read, but not written
2. Same test case with emptydir
3. Create a pod with one host volume but different mount permissions for each container. One container should be read only, the other should be read-write

This will replace the similar test cases on Linux:

```
[sig-storage] EmptyDir volumes  should support (root,0644,default) [NodeConformance] [Conformance]
[sig-storage] EmptyDir volumes  should support (root,0666,default) [NodeConformance] [Conformance]
[sig-storage] EmptyDir volumes  should support (root,0666,tmpfs) [NodeConformance] [Conformance]
```

/kind test 
/sig windows

**Comment by andyzhangx** [comment]
volume on windows does not support `read-only` now, it's all mounted as readwrite

**Comment by astrieanna** [comment]
Hey @PatrickLang, I have a couple questions about the test cases needed here.

>Create a pod with a host mapped volume mapped read-only. Files should be able to be read, but not written

By "host mapped volume", do you mean [hostpath](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath)?

>Create a pod with one host volume but different mount permissions for each container. One container should be read only, the other should be read-write

After reading through the[ volume docs page](https://kubernetes.io/docs/concepts/storage/volumes), I'm still not sure how to mount a volume with different permissions in different containers. Could you point to an example of how this would be configured?

**Comment by andyzhangx** [comment]
@astrieanna It's `EmptyDir` test case.
Currently when mounting windows a windows dir, `ro` options is not respected. Need to find a way to make a directory as ReadOnly. 
So this case may be dependent on https://github.com/kubernetes/kubernetes/issues/63051

**Comment by vyta** [comment]
I will be picking this work item up.

**Comment by vyta** [comment]
I'd like to pick this work item up

**Comment by vyta** [comment]
Since `ro` is still unimplemented for mounting windows dir, should we implement a skip on the test or put this on hold? @PatrickLang

**Comment by PatrickLang** [comment]
Windows can't do a read-only root, but it can do a read-only volume mount

**Comment by benmoss** [comment]
This was resolved in #74085
/close
